### ISSUE TITLE
Better support for connection loss/reconnect

### ISSUE THREAD

**Issue by ndarilek** [home-assistant/home-assistant-android#165]
First off, I'll mention that I'm a blind TalkBack user, so I'm not sure if there's some obvious indicator or control I'm missing.

I'm using Home Assistant over a Wireguard connection. Last weekend was my first opportunity to take the setup for a spin, and I started noticing odd issues. Essentially, I think most of them boiled down to the app losing, and not regaining, its connection to the server. I'm definitely experiencing flakiness with Wireguard as well, but my HA connection seemed to routinely get into a state where it wasn't connected, there was no obvious/accessible indicator that it wasn't (other than the text "Connection lost, reconnecting" at the bottom of my screen which never goes away) and I'd often see outdated information in the interface. At one point, I killed and restarted the app entirely, only to discover that my server's domain name wasn't responding, and the Wireguard connection needed to be restarted.

I wonder if we can have a better, and accessible, means of knowing that HA is connected/reconnected/disconnected? Maybe that's a bug for HA proper? Also, would a _Refresh_ button in the sidebar be possible so at the very least I can force the interface to reload and know for sure that I'm connected/having DNS issues?

Thanks for a great app.

**Comment by balloob** [comment]
The UI in the app is provided by Home Assistant itself. The code for that is in the Polymer repo: https://github.com/home-assistant/home-assistant-polymer/. 

If the "Connection lost, reconning" toast is not going away, it means the connection was never re-established. That seems like a connection issue.

The frontend also notifies the app of it's connection status, but that is not being used yet ([docs](https://developers.home-assistant.io/docs/en/frontend_external_bus.html#connection-status-update-connection-status)).

**Comment by ndarilek** [comment]
Got it. It doesn't vanish for me with either Orca under Linux or 
TalkBack under Android, so I suspect it isn't being hidden from an a11y 
perspective. I'll investigate and file a bug in the UI repo.


But would it be possible to add a Refresh button somewhere? I'm not sure 
how robust reconnection handling is.


I did just disable battery optimization, so perhaps the app will more 
aggressively attempt to reconnect. If that's the case, a Refresh button 
may not be necessary and I'll report back after a few days of testing.


Thanks.

**Comment by ronsum232** [comment]
I find that with the app open and active, a HA restart will show the app's "Connection lost, reconnecting" toast, but it will never reconnect. The app must be closed and reopened.

**Comment by andriej** [comment]
Or it takes long, much longer than for webview as I feel it.

**Comment by JBassett** [comment]
I have never had it not reconnect.  I agree it might take a while, but as @balloob said that would be a frontend issue not an application issue.

**Comment by ronsum232** [comment]
You don't have the issue, so there is no issue... understood.

**Comment by JBassett** [comment]
If that's how my comment sounded I apologize.  The application itself doesn't handle the reconnection, that's all handled by Home Assistants front end.  If you are having trouble with it this is the correct repository to report the issue: https://github.com/home-assistant/home-assistant-polymer

**Comment by Erriez** [comment]
Related to https://github.com/home-assistant/frontend/issues/7732.
